schema/shared/src/main/scala/zio/blocks/schema/DynamicPatch.scala

```diff
@@ -0,0 +1,817 @@
+          Left(SchemaError.TypeMismatch("Double", other.getClass.getSimpleName))
+      }
+    }
+  }
+
+  final case class FloatDelta(delta: Float) extends PrimitiveOp {
+    def applyTo(value: DynamicValue, mode: PatchMode): Either[SchemaError, DynamicValue] = {
+      value match {
+        case DynamicValue.Primitive(PrimitiveValue.Float(n)) =>
+          Right(DynamicValue.Primitive(PrimitiveValue.Float(n + delta)))
+        case other =>
+          Left(SchemaError.TypeMismatch("Float", other.getClass.getSimpleName))
+      }
+    }
+  }
+
+  final case class ShortDelta(delta: Short) extends PrimitiveOp {
+    def applyTo(value: DynamicValue, mode: PatchMode): Either[SchemaError, DynamicValue] = {
+      value match {
+        case DynamicValue.Primitive(PrimitiveValue.Short(n)) =>
+          Right(DynamicValue.Primitive(PrimitiveValue.Short((n + delta).toShort)))
+        case other =>
+          Left(SchemaError.TypeMismatch("Short", other.getClass.getSimpleName))
+      }
+    }
+  }
+
+  final case class ByteDelta(delta: Byte) extends PrimitiveOp {
+    def applyTo(value: DynamicValue, mode: PatchMode): Either[SchemaError, DynamicValue] = {
+      value match {
+        case DynamicValue.Primitive(PrimitiveValue.Byte(n)) =>
+          Right(DynamicValue.Primitive(PrimitiveValue.Byte((n + delta).toByte)))
+        case other =>
+          Left(SchemaError.TypeMismatch("Byte", other.getClass.getSimpleName))
+      }
+    }
+  }
+
+  final case class BigIntDelta(delta: BigInt) extends PrimitiveOp {
+    def applyTo(value: DynamicValue, mode: PatchMode): Either[SchemaError, DynamicValue] = {
+      value match {
+        case DynamicValue.Primitive(PrimitiveValue.BigInt(n)) =>
+          Right(DynamicValue.Primitive(PrimitiveValue.BigInt(n + delta)))
+        case other =>
+          Left(SchemaError.TypeMismatch("BigInt", other.getClass.getSimpleName))
+      }
+    }
+  }
+
+  final case class BigDecimalDelta(delta: BigDecimal) extends PrimitiveOp {
+    def applyTo(value: DynamicValue, mode: PatchMode): Either[SchemaError, DynamicValue] = {
+      value match {
+        case DynamicValue.Primitive(PrimitiveValue.BigDecimal(n)) =>
+          Right(DynamicValue.Primitive(PrimitiveValue.BigDecimal(n + delta)))
+        case other =>
+          Left(SchemaError.TypeMismatch("BigDecimal", other.getClass.getSimpleName))
+      }
+    }
+  }
+
+  /** String edit operations (LCS-based) */
+  final case class StringEdit(ops: Vector[StringOp]) extends PrimitiveOp {
+    def applyTo(value: DynamicValue, mode: PatchMode): Either[SchemaError, DynamicValue] = {
+      value match {
+        case DynamicValue.Primitive(PrimitiveValue.String(s)) =>
+          val result = StringOp.apply(s, ops)
+          Right(DynamicValue.Primitive(PrimitiveValue.String(result)))
+        case other =>
+          Left(SchemaError.TypeMismatch("String", other.getClass.getSimpleName))
+      }
+    }
+  }
+}
+
+/**
+ * String edit operations for LCS-based diffing.
```

Comment:
Obviously anything this complex must have its own test suite. ALso, it's super dirty to pollute the top-level namespace with all these ops. I would have ALL ops stored in the companion object of `DynamicPatch`.